exercises/bob/bob_test.rb

```diff
@@ -2,133 +2,184 @@
-    assert_equal 'Whoa, chill out!', bob.hey(remark), feedback(remark)
+    remark = %Q(ZOMG THE %^*@#$(*^ ZOMBIES ARE COMING!!11!!1!)
+    assert_equal 'Whoa, chill out!', Bob.hey(remark), %Q(Bob hears ZOMG THE %^*@#$(*^ ZOMBIES ARE COMING!!11!!1!, and..)
   end
 
   def test_shouting_with_no_exclamation_mark
     skip
-    remark = 'I HATE YOU'
-    assert_equal 'Whoa, chill out!', bob.hey(remark), feedback(remark)
+    remark = %Q(I HATE YOU)
+    assert_equal 'Whoa, chill out!', Bob.hey(remark), %Q(Bob hears I HATE YOU, and..)
   end
 
   def test_statement_containing_question_mark
     skip
-    remark = 'Ending with ? means a question.'
-    assert_equal 'Whatever.', bob.hey(remark), feedback(remark)
+    remark = %Q(Ending with ? means a question.)
+    assert_equal 'Whatever.', Bob.hey(remark), %Q(Bob hears Ending with ? means a question., and..)
+  end
+
+  def test_non_letters_with_question
+    skip
+    remark = %Q(:) ?)
+    assert_equal 'Sure.', Bob.hey(remark), %Q(Bob hears :) ?, and..)
   end
 
   def test_prattling_on
     skip
-    remark = 'Wait! Hang on. Are you going to be OK?'
-    assert_equal 'Sure.', bob.hey(remark), feedback(remark)
+    remark = %Q(Wait! Hang on. Are you going to be OK?)
+    assert_equal 'Sure.', Bob.hey(remark), %Q(Bob hears Wait! Hang on. Are you going to be OK?, and..)
   end
 
   def test_silence
     skip
-    remark = ''
-    assert_equal 'Fine. Be that way!', bob.hey(remark), feedback(remark)
+    remark = %Q()
+    assert_equal 'Fine. Be that way!', Bob.hey(remark), %Q(Bob hears , and..)
   end
 
   def test_prolonged_silence
     skip
-    remark = ' ' * rand(1..10)
-    assert_equal 'Fine. Be that way!', bob.hey(remark), feedback(remark)
+    remark = %Q(          )
+    assert_equal 'Fine. Be that way!', Bob.hey(remark), %Q(Bob hears           , and..)
   end
 
-  def test_alternate_silences
+  def test_alternate_silence
     skip
-    remark = "\t" * rand(1..10)
-    assert_equal 'Fine. Be that way!', bob.hey(remark), feedback(remark)
+    remark = %Q(										)
+    assert_equal 'Fine. Be that way!', Bob.hey(remark), %Q(Bob hears 										, and..)
   end
 
-  def test_on_multiple_line_questions
+  def test_multiple_line_question
     skip
-    remark = %(
+    remark = %Q(
 Does this cryogenic chamber make me look fat?
 no)
-    assert_equal 'Whatever.', bob.hey(remark), feedback(remark)
+    assert_equal 'Whatever.', Bob.hey(remark), %Q(Bob hears 
+Does this cryogenic chamber make me look fat?
+no, and..)
+  end
+
+  def test_starting_with_whitespace
+    skip
+    remark = %Q(         hmmmmmmm...)
+    assert_equal 'Whatever.', Bob.hey(remark), %Q(Bob hears          hmmmmmmm..., and..)
+  end
+
+  def test_ending_with_whitespace
+    skip
+    remark = %Q(Okay if like my  spacebar  quite a bit?   )
+    assert_equal 'Sure.', Bob.hey(remark), %Q(Bob hears Okay if like my  spacebar  quite a bit?   , and..)
+  end
+
+  def test_other_whitespace
+    skip
+    remark = %Q(
+
 	)
```

Comment:
Using inspect should also make the silly whitespace characters usefully visible.